plugins/cros_ec-usb/fu-cros_ec-usb-common.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2020 Benson Leung <bleung@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-cros_ec-usb-common.h"
+
+gboolean
+fu_cros_ec_usb_parse_version (const gchar *version_raw,
+			      struct cros_ec_version *version,
+			      GError **error)
+{
+	g_auto(GStrv) v_split = NULL;
+	g_auto(GStrv) marker_split = NULL;
+	g_auto(GStrv) triplet_split = NULL;
+
+	if (NULL == version_raw || 0 == strlen (version_raw)) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INTERNAL,
+				     "No version string to parse.");
+		return FALSE;
+	}
+
+	/* sample version string: cheese_v1.1.1755-4da9520 */
+	v_split = g_strsplit (version_raw, "_v", 2);
+	if (2 > g_strv_length (v_split)) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INTERNAL,
+				     "Version marker not found.");
+		return FALSE;
+	}
+	marker_split = g_strsplit_set (v_split[1], "-+", 2);
+	if (2 > g_strv_length (marker_split)) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INTERNAL,
+			     "Hash marker not found: %s", v_split[1]);
+		return FALSE;
+	}
+	triplet_split = g_strsplit_set (marker_split[0], ".", 3);
+	if (3 > g_strv_length (triplet_split)) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INTERNAL,
+			     "Improper version triplet: %s", marker_split[0]);
+		return FALSE;
+	}
+	g_strlcpy (version->triplet, marker_split[0], 32);
+	if (0 == g_strlcpy (version->boardname, v_split[0], 32)) {
```

Comment:
Would g_strndup() be more suitable here? It's hardly fast path.

Reply:
Hmm. I've allocated enough space in cros_ec_version for these strings, so a straight up g_strlcpy saves me from having to free it later separately.